agent/structs/structs.go

```diff
@@ -1000,11 +1006,35 @@ func (ns *NodeService) compoundID(preferName bool) ServiceID {
 }
 
 func (ns *NodeService) CompoundServiceID() ServiceID {
-	return ns.compoundID(false)
+	id := ns.ID
+	if id == "" {
+		id = ns.Service
+	}
+
+	// copy the ent meta and normalize it
+	entMeta := ns.EnterpriseMeta
+	entMeta.Normalize()
+
+	return ServiceID{
+		ID:             id,
+		EnterpriseMeta: entMeta,
+	}
 }
 
-func (ns *NodeService) CompoundServiceName() ServiceID {
-	return ns.compoundID(true)
```

Comment:
I believe you can delete `ns.compoundID` now, since we've expanded the function here.

Reply:
True, done